core/sail/federation/src/main/java/org/eclipse/rdf4j/sail/federation/optimizers/EmptyPatternOptimizer.java

```diff
@@ -58,7 +71,10 @@ public void meet(StatementPattern node)
 		Value obj = node.getObjectVar().getValue();
 		Resource[] ctx = getContexts(node.getContextVar());
 		for (RepositoryConnection member : members) {
-			if (member.hasStatement(subj, pred, obj, true, ctx)) {
+			RepositoryBloomFilter bloomFilter = MoreObjects.firstNonNull(
```

Comment:
As said above: if the output of the `bloomFilters` function is an `Optional<RepositoryBloomFilter>` you can replace this with:

    RepositoryBloomFilter bloomFilter = bloomFilters.apply(member.getRepository())
                                                                        .or(AccurateRepositoryBloomFilter.INCLUDE_INFERRED_INSTANCE);